README.md

```diff
@@ -1,15 +1,55 @@
 <a href="https://codeclimate.com/github/ASAAR/SE2-KaggleComp"><img src="https://codeclimate.com/github/ASAAR/SE2-KaggleComp/badges/gpa.svg" /></a> <a href="https://codeclimate.com/github/ASAAR/SE2-KaggleComp"><img src="https://codeclimate.com/github/ASAAR/SE2-KaggleComp/badges/issue_count.svg" /></a>
 
-# SE2-KaggleComp
+# SE2 Machine Learning Framework
+
 Software Engineering 2 class kaggle competition repo. We'll also be building a bit of a framework to help make doing these easier in the future.
 
-## Setup Instructions
-**You must be using a unix system for this software to work!**
+## Overview
+
+### Purpose
+To aid developers using machine learning algorithms in finding the best algorithms and optimal configurations for their specific situation. This is accomplished by recording as much information on a certain model as the developer wants, and then analyzing all the data to find which algorithms work best on a dataset and with what settings to they work best.
+
+### Features
+- Automatically applies several of a number of machine learning algorithms against the input data based on the settings it is given
+- Has a large number of tests that make sure algorithms included still run and haven't become outdated
+- Can be put in validation or application mode (train/test mode)
+- Record results from a machine learning algorithm test
+- Saves results in a Firebase Database
+- (Coming Soon) View results in a WebUI
+- (Coming Soon) Analyze data from results in the WebUI
+
+### Terminology
+
+#### Test
+Unless otherwise stated, when we say test we mean a way of determining if an algorithm works. (As opposed to testing of the actual code, etc..)
+
+#### Test Data
+Test data is data that is unlabeled, in other words it does not have a column or label which represents the target that a model is trying to predict. So if a model predicts housing prices, "test data" will not have the housing prices listed.
+
+#### Train Data
+Training data on the other hand does have the target column, because the model has to use that column to be trained.
+
+#### Result Record
+
+All the information about the test including hyper-parameters of the model used, information on the test data, results of test, etc.
 
-###### Setting up Ubuntu Bash for Windows
-[Use this tutorial](http://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/)
+## Getting Started
 
-###### Setting up machine learning libraries for Python
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system.
```

Comment:
These instructions will get you a copy of the **entire** project up and running on your local machine for development and testing purposes. See deployment for notes on how to deploy the project on a live system. 
If however, you only wish to run some modules, please see that individual modules installation instructions for setting up.